pkg/pagination/pagination.go

```diff
@@ -93,6 +93,14 @@ func (pb *Bag) Next(pageToken string) error {
 
 // Next pops the current token, and pushes a copy of it with an updated page token.
 func (pb *Bag) NextToken(pageToken string) (string, error) {
+	// assume that `pb` was passed an empty token
+	if pb.currentState == nil {
```

Comment:
if `pageToken` is `""` here, should we return `"", nil`?  eg, the nextToken is empty?

Reply:
if `pageToken` is empty, the same state is returned per line 109. During my tests with baton-aws I didn't fall into infinite loops